### PR TITLE
[NUP] A06 Vsevolod Vaskin

### PR DESCRIPTION
**Pull request by Seva-Vaskin**

### CODE REVIEW

File: src/Expr.lama [hunk 1]
```diff
@@ -101,9 +94,80 @@ fun evalList (c, exprs) {
...
+  }
+  foldl(myAddName, state, zip(args, vals))
+}
+
 fun eval (c@[s, w], expr) {
-  failure ("evalExpr not implemented\n")
+  case expr of
+      Var (x) -> [c, lookup(s, x)]
+    | Ref (x) -> [c, x]
+    | Const (x) -> [c, x]
```
Comment (e2e4b6b7) [inline src/Expr.lama:117]: wrap in Var

File: src/X86.lama [hunk 2]
```diff
@@ -410,40 +410,151 @@ fun suffix (op) {
...
+                          [env, code
+                            <+> pushRegisters(registers)
+                            <+> pushedArguments
+                            <+ Call(fLabel)
+                            <+ Binop("+", L(wordSize * nArgs), esp)
+                            <+> popRegisters(reverse(registers))
+                            <+> move(eax, res)
+                          ]
+                      esac
+                  esac
```
Comment (e2e4b6b7) [inline src/X86.lama:555]: ❤️

**Review by e2e4b6b7 — CHANGES_REQUESTED** [review]

**Comment by e2e4b6b7** [comment]
ping

**Comment by Seva-Vaskin** [comment]
Fixed

**Review by e2e4b6b7 — APPROVED** [review]

**Pull request closed without merging**